spec/requests/api/v3/statements_spec.rb

```diff
@@ -1,13 +1,66 @@
 require "rails_helper"
 
 RSpec.describe "Statements endpoint", type: "request" do
+  let(:current_lead_provider) { create(:lead_provider) }
+
   describe "GET /api/v3/statements" do
     context "when authorized" do
-      it "returns 200 - ok" do
-        api_get("/api/v3/statements")
+      context "when 2 statements exist for current_lead_provider" do
+        let!(:statement1) { create(:statement, lead_provider: current_lead_provider) }
+        let!(:statement2) { create(:statement, lead_provider: current_lead_provider) }
+
+        before do
+          create(:statement, lead_provider: create(:lead_provider, name: "Another lead provider"))
+        end
+
+        it "returns 2 statements" do
+          api_get("/api/v3/statements")
 
-        expect(response.status).to eq 200
-        expect(response.headers["Content-Type"]).to include("application/json")
+          expect(response.status).to eq 200
+          expect(response.content_type).to eql("application/json")
+          expect(parsed_response["data"].size).to eq(2)
+          expect(parsed_response["data"][0]["id"]).to eq(statement1.id)
+          expect(parsed_response["data"][1]["id"]).to eq(statement2.id)
+        end
+      end
+
+      context "when not statements exist" do
+        it "returns empty" do
+          api_get("/api/v3/statements")
+
+          expect(response.status).to eq 200
+          expect(parsed_response["data"].size).to eq(0)
+        end
+      end
+
+      describe "filtering" do
+        describe "by cohort" do
+          let(:cohort_2023) { create(:cohort, start_year: 2023) }
+          let(:cohort_2024) { create(:cohort, start_year: 2024) }
+          let(:cohort_2025) { create(:cohort, start_year: 2025) }
+
+          it "returns statements for the specified cohort" do
+            create(:statement, lead_provider: current_lead_provider, cohort: cohort_2023)
+            create(:statement, lead_provider: current_lead_provider, cohort: cohort_2024)
+            create(:statement, lead_provider: current_lead_provider, cohort: cohort_2025)
+
+            api_get("/api/v3/statements", params: { filter: { cohort: "2023,2024" } })
+
+            expect(parsed_response["data"].size).to eq(2)
+          end
+        end
+
+        describe "by updated_since" do
+          it "returns statements updated since the specified date" do
+            create(:statement, lead_provider: current_lead_provider, updated_at: 6.hours.ago)
+            create(:statement, lead_provider: current_lead_provider, updated_at: 3.hours.ago)
+            create(:statement, lead_provider: current_lead_provider, updated_at: 1.hour.ago)
+
+            api_get("/api/v3/statements", params: { filter: { updated_since: 4.hours.ago } })
+
+            expect(parsed_response["data"].size).to eq(2)
+          end
+        end
```

Comment:
Like with #1215, I think this is going to cause overlap down the line.

The responsibility of the controller here is to receive a request with some params, make sure the params are sound and pass them into the query service.

In my opinion that's all we should be testing here.

Reply:
> Also, if we don't test some filtering, how can we be sure that we are even plugged to the right service with the right parameters?

Reply:
I agree with @pmanrubia request specs exercise the full stack and can drive full integration testing. Keeping them simple and testing the happy path seems fair to me, as long as we don't go too far with them.